TPP.Core/Commands/Definitions/OperatorCommands.cs

```diff
@@ -183,18 +184,38 @@ public async Task<CommandResult> TransferBadge(CommandContext context)
 
         public async Task<CommandResult> CreateBadge(CommandContext context)
         {
-            (User recipient, PkmnSpecies species, Optional<PositiveInt> amountOpt) =
-                await context.ParseArgs<AnyOrder<User, PkmnSpecies, Optional<PositiveInt>>>();
+            (User recipient, PkmnSpecies species, Optional<PositiveInt> amountOpt, Optional<string> formOpt, Optional<Shiny> shinyOpt) =
+                await context.ParseArgs<AnyOrder<User, PkmnSpecies, Optional<PositiveInt>, Optional<string>, Optional<Shiny>>>();
             int amount = amountOpt.Map(i => i.Number).OrElse(1);
-
+            int form = PkmnForms.pokemonHasForms(species) ? 0 : 1; // default to the first listed form if form is unspecified
```

Comment:
I don't understand the comment. Why `1`? Isn't `0` an acceptable default for both cases?

Reply:
Forms are 1-indexed. The idea is that 0 means forms don't exist for the species, and positive ints are used for the forms of species that have forms. E.G. a form 0 shellos is invalid, and a form 1 caterpie is also invalid.

Reply:
according to this table: https://pixelmonmod.com/wiki/Form_indices they are 0-indexed, and shellos 0 is shellos east for example

Reply:
I think it's beneficial to have the form numbers we use be completely synchronized with the official form indexes, so we don't need to remember to apply some mapping every time we generate a badge with a form number

Reply:
I wasn't aware this existed. If this matches the official specification I'll just use the values here.

Reply:
no idea if that website is "official" in all cases, but official form numbers do exist. I couldn't find a more "official" list